velox/functions/sparksql/registration/RegisterMath.cpp

```diff
@@ -120,6 +122,7 @@ void registerMathFunctions(const std::string& prefix) {
   registerBinaryNumeric<CheckedSubtractFunction>({prefix + "checked_subtract"});
   registerBinaryNumeric<CheckedMultiplyFunction>({prefix + "checked_multiply"});
   registerBinaryNumeric<CheckedDivideFunction>({prefix + "checked_divide"});
+  registerFunctionCallToSpecialForm("factorial",std::make_unique<FactorialCallToSpecialForm>());
```

Comment:
Since in Spark the input type for 'Factorial' is IntegerType and the output type is 'LongType', it appears that this function does not require registration as a special form. We could register it as a simple function.
(Special forms are for those functions whose return type cannot be fully resolved from the argument types)